From 1160c25898b84f5c8477c3ab08c78a24c3b48b2e Mon Sep 17 00:00:00 2001
From: tenzap <fabstz-it@yahoo.fr>
Date: Tue, 2 May 2023 10:04:37 +0200
Subject: [PATCH 1/2] use setWindowFlags() instead of setWindowFlag on Qt<5.9

Fixes:
src/gui/CloneDialog.cpp:32:5: error: use of undeclared identifier 'setWindowFlag'
    setWindowFlag(Qt::WindowContextHelpButtonHint, false);
    ^

void QWidget::setWindowFlag(Qt::WindowType flag, bool on = true)
--> This function was introduced in Qt 5.9.
---
 src/gui/CloneDialog.cpp        | 4 ++++
 src/gui/DatabaseOpenDialog.cpp | 4 ++++
 src/gui/TotpSetupDialog.cpp    | 4 ++++
 3 files changed, 12 insertions(+)

diff --git a/src/gui/CloneDialog.cpp b/src/gui/CloneDialog.cpp
index 4ba5856b..72199ae5 100644
--- src/gui/CloneDialog.cpp
+++ src/gui/CloneDialog.cpp
@@ -29,7 +29,11 @@ CloneDialog::CloneDialog(DatabaseWidget* parent, Database* db, Entry* entry)
     m_ui->setupUi(this);
 
     window()->layout()->setSizeConstraint(QLayout::SetFixedSize);
+#if QT_VERSION >= QT_VERSION_CHECK(5, 9, 0)
     setWindowFlag(Qt::WindowContextHelpButtonHint, false);
+#else
+    setWindowFlags(windowFlags() & ~Qt::WindowContextHelpButtonHint);
+#endif
     setAttribute(Qt::WA_DeleteOnClose);
 
     connect(m_ui->buttonBox, SIGNAL(rejected()), SLOT(close()));
diff --git a/src/gui/DatabaseOpenDialog.cpp b/src/gui/DatabaseOpenDialog.cpp
index acf90504..366f7a9c 100644
--- src/gui/DatabaseOpenDialog.cpp
+++ src/gui/DatabaseOpenDialog.cpp
@@ -36,7 +36,11 @@ DatabaseOpenDialog::DatabaseOpenDialog(QWidget* parent)
 {
     setWindowTitle(tr("Unlock Database - KeePassXC"));
     setWindowFlags(Qt::Dialog);
+#if QT_VERSION >= QT_VERSION_CHECK(5, 9, 0)
     setWindowFlag(Qt::WindowContextHelpButtonHint, false);
+#else
+    setWindowFlags(windowFlags() & ~Qt::WindowContextHelpButtonHint);
+#endif
 #ifdef Q_OS_LINUX
     // Linux requires this to overcome some Desktop Environments (also no Quick Unlock)
     setWindowFlags(windowFlags() | Qt::WindowStaysOnTopHint);
diff --git a/src/gui/TotpSetupDialog.cpp b/src/gui/TotpSetupDialog.cpp
index 875dfcfd..d33d9ce3 100644
--- src/gui/TotpSetupDialog.cpp
+++ src/gui/TotpSetupDialog.cpp
@@ -29,7 +29,11 @@ TotpSetupDialog::TotpSetupDialog(QWidget* parent, Entry* entry)
 {
     m_ui->setupUi(this);
     setAttribute(Qt::WA_DeleteOnClose);
+#if QT_VERSION >= QT_VERSION_CHECK(5, 9, 0)
     setWindowFlag(Qt::WindowContextHelpButtonHint, false);
+#else
+    setWindowFlags(windowFlags() & ~Qt::WindowContextHelpButtonHint);
+#endif
     setFixedSize(sizeHint());
 
     connect(m_ui->buttonBox, SIGNAL(rejected()), SLOT(close()));
-- 
2.30.2

